### PR TITLE
#184797946 Admin CRUD product

### PR DESCRIPTION
**Pull request by bergad37**
### What does this PR do?
Admin should be able to perfom  CRUD operations on  products in stock

### Description of Task to be completed?

- Admin can crud the product
- Admin can delete, update, read, and add a product in the stock.

### How should this be manually tested?

- Clone the repository using git clone [link](https://github.com/atlp-rwanda/e-comm-team-techsmith-bn)
- Install node packages that the application depends on: `npm i`
- Checkout to the branch using `git checkout ft-admin-crud-product-184797946`
- 
- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in the application's root directory.
- Start the application development server: `npm run dev` ,and test admin routes

## Any background context you want to provide?

- The user have full access to the products in the stock regardless the owner of the product

## What are the relevant pivotal tracker/Trello stories?
[#184797946](https://www.pivotaltracker.com/story/show/184797946)

## Screenshots 
Screen Shot 2023-04-23 at 23 55 10

Screen Shot 2023-04-23 at 11 11 54
Screen Shot 2023-04-23 at 15 57 10

Screen Shot 2023-04-23 at 15 55 36

### CODE REVIEW

**Review by irakevine — COMMENTED** [review]
You removed that line 24 that I think is unnecessary

File: src/docs/Admin/getASingleProduct.js
Comment (frankflej) [inline src/docs/Admin/getASingleProduct.js:1]: I think you would make some changes in the description and operationId to show that you are getting a single product

File: src/docs/Product/getAsingleProduct.js
Comment (frankflej) [inline src/docs/Product/getAsingleProduct.js:1]: The same applies here!!!

**Comment by bergad37** [comment]
Thanks, line 24 has been removed!

**Review by frankflej — COMMENTED** [review]
Check in `Docs/Admin/getASingleProduct` and ` Docs/Product/getASingleProduct` in operationId and description the messages you are passing there are incorrect please make some modifications.

**Review by nishimweprince — APPROVED** [review]
Excellent work!! I also appreciate that you considered my previous comments.

**Review by niyontwali — COMMENTED** [review]
Please resolve all the conversions.

**Review by cyubahiro367 — APPROVED** [review]
you did a great work kudos

**Review by jcharles921 — APPROVED** [review]
You really did an amazing job from the functions in your CRUD controllers to your tests✨✨✨. Your work is meticulous , you can see it from your Perfectly done Swagger Docs ...

**Comment by bergad37** [comment]
I have made changes in the new PR you can check!

**Pull request closed without merging**